File: iOS/DataPool/Sources/DataPool/Application/GetOpenPoolLayoutsUseCase.swift
import Core

public class GetOpenPoolLayoutsUseCase {
    private let poolLayoutRepository: PoolLayoutRepository

    public init(poolLayoutRepository: PoolLayoutRepository) {
        self.poolLayoutRepository = poolLayoutRepository
    }

    public func execute(
        next: String? = nil,
        searchText: String? = nil
    ) async -> Result<CursorPage<PoolLayout>, Error> {
        return await poolLayoutRepository.getOpenPoolLayouts(next: next, searchText: searchText)
    }
}

File: Android/data/pool/src/main/java/com/felipearpa/tyche/data/pool/application/GetOpenPoolLayoutsUseCase.kt
package com.felipearpa.tyche.data.pool.application

import com.felipearpa.tyche.data.pool.domain.PoolLayoutRepository

class GetOpenPoolLayoutsUseCase internal constructor(private val poolLayoutRepository: PoolLayoutRepository) {
    suspend fun execute(next: String?, searchText: String?) =
        poolLayoutRepository.getOpenPoolLayouts(next = next, searchText = searchText)
}
